gitlab/http_backends/requests_backend.py

```diff
@@ -0,0 +1,15 @@
+from typing import Optional
+
+import requests
+
+
+class RequestsBackend:
+
+    _client: requests.Session
+
+    def __init__(self, session: Optional[requests.Session] = None) -> None:
+        self._client = session or requests.Session()
```

Comment:
Same question as above, I believe we can add the hint directly in the assignment? mypy seems to be happy with the move.

```suggestion
    def __init__(self, session: Optional[requests.Session] = None) -> None:
        self._client: requests.Session = session or requests.Session()
```

Reply:
Same comment as above